bdb/lite.c

```diff
@@ -377,8 +410,7 @@ static int bdb_lite_fetch_keys_int(bdb_state_type *bdb_state, tran_type *tran,
         return -1;
     }
 
-    db = bdb_state->dbp_data[0][0];
-    rc = db->cursor(db, tid, &dbcp, 0);
+    rc = get_cursor_lite(bdb_state, tran, &dbcp);
```

Comment:
can you check below codepaths? seems we return from this function on some errors without closing the cursor.

Reply:
AH THANK YOU ADI!